timechart/displays/main_display.py

```diff
@@ -702,7 +702,11 @@ def add_y_channel(self, pv_name, curve_name, color, line_style=Qt.SolidLine,
         self.generate_pv_controls(pv_name, color)
 
         self.enable_chart_control_buttons()
-        self.app.add_connection(curve.channel)
+        try:
+            self.app.add_connection(curve.channel)
+        except AttributeError:
+            # these methods are not needed on future versions of pydm
```

Comment:
@hhslepicka Can future versions of PyDM still maintain the add_connection method for backward compatibility? Requiring developers to keep a note (written/mental) whenever they call a method is a little messy.

Reply:
@hmbui i don't understand your comment about keeping a note (written/mental).
If you look at PyDM, the code is still there https://github.com/slaclab/pydm/blob/master/pydm/application.py#L483-L493.
What @tacaswell added here is a protection for when this method gets removed.
We marked the method for deprecation which means that in some time in the future we will remove it... 
If you depend on PyDM for your application it is your responsability to keep up with the versions or add more strict dependency rules to avoid users to use your software with a new version.

Reply:
@hhslepicka, PyDMConnection.connect() isn't available:
https://github.com/slaclab/pydm/blob/master/pydm/data_plugins/plugin.py

When will it be?

If the warning in PyDMApplication.add_connection() says "use PyDMConnection.connect()", eventually, add_connection() has to be replaced with PyDMConnection.connect().  In any case, AttributeError shouldn't be ignored.

So, @tacaswell should not have a pass on AttributeError exception handling. His note for this PR is that "Current pydm master branch does not have these methods", but @hhslepicka is saying the method is still available. This is yet another confusion.

Reply:
First, the warning at add_connection() isn't clear. It says "use PyDMConnection.connect()". What it should say is it's no longer needed.

Then, it's a matter of how to proceed with the PyDM change properly. I don't like to catch AttributeError exception in multiple places. Either we let this exception be raised and alert the user that they must upgrade PyDM, or PyDM should keep add_connection() available until most PyDM users have upgraded to version 1.6.0.

None of that actually matters since I'll be updating requirements.txt to require pydm>=1.6.1 for TimeChart anyway, I think all the add_connection() calls should just be removed from the TimeChart code base. Do you agree, @hhslepicka?

Reply:
Some more discussion here: https://github.com/slaclab/timechart/issues/35

@hmbui We have external collaborators who want to use `timechart`, but have a basic `QApplication` so this is needed now. I concur with both @hhslepicka and your above conclusion that the calls should to `add_connection` should just be removed entirely and we specify in the `requirements` file and `Conda` recipe that you need to use `pydm>=1.6`

Reply:
@teddyrendahl is right! We are removing all the dependencies on the `PyDMApplication` and `PyDMMainWindow` from the widgets.
This allows users to use PyDM widgets into their own Qt applications without the need of those 2 pieces that most of the time are not necessary (unless you want to use the navigation).
Given that and the fact that some time in the future the method will be removed, it totally makes sense to add the code changes proposed by Tom to protect against failure when we deprecate the method as well as allow for users to use TimeChart on their application even if they don't make use of `PyDMApplication` and `PyDMMainWindow`.

Reply:
Good discussion.

Since I'd like to have the PyDM requirement changed in the same release with the StripTool format support, for now, I will approve this change so that Tom can push and make the code work for non-PyDM users for the time being. When the StripTool format support is approved, I'll have the PyDM's PyPi requirement also merged, will remove the add_connection() usages.